### PR TITLE
run-make-check,cmake: use clang in run-make-check.sh

### PR DESCRIPTION
**Pull request by tchaikov**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by tchaikov** [comment]
bumping up the fio version addresses following compiling error:
```
src/test/fio/CMakeFiles/fio_ceph_objectstore.dir/fio_ceph_objectstore.cc.o.d -o src/test/fio/CMakeFiles/fio_ceph_objectstore.dir/fio_ceph_objectstore.cc.o -c ../src/test/fio/fio_ceph_objectstore.cc
In file included from ../src/test/fio/fio_ceph_objectstore.cc:26:
In file included from src/fio/fio.h:18:
In file included from src/fio/thread_options.h:6:
In file included from src/fio/options.h:8:
src/fio/parse.h:128:13: error: arithmetic on a pointer to void
        return ret + offset;
               ~~~ ^
1 error generated.
```

**Comment by tchaikov** [comment]
```
DEBUG 2021-08-14 02:38:09,380 [shard 0] seastore - OTree::Seastore(0x601000858140): get root: trigger eagain

unittest-staged-fltree: ../src/crimson/os/seastore/onode_manager/staged-fltree/node.cc:420: crimson::os::seastore::onode::Node::load_root(crimson::os::seastore::onode::context_t, crimson::os::seastore::onode::RootNodeTracker&)::<lambda(auto:115&&)> [with auto:115 = std::unique_ptr<crimson::os::seastore::onode::Super>]: Assertion `_super' failed.
...
The following tests FAILED:
	235 - unittest-staged-fltree (Child aborted)
```

**Comment by tchaikov** [comment]
jenkins test make check

**Comment by tchaikov** [comment]
jenkins test make check

**Comment by tchaikov** [comment]
jenkins test make check

**Comment by wjwithagen** [comment]
@tchaikov 

Just out of curriosity.
When compiling with Clang, does that also imply the the llvm linker in used.
Or is linking still with gnu-bintools ld??

**Comment by tchaikov** [comment]
jenkins test make check

**Comment by tchaikov** [comment]
@wjwithagen 

the linker from GNU Binutils is used. see

```
jenkins+ 2435292  106  0.2 686520 684644 ?       R    04:03   0:04 /usr/bin/ld -z relro --hash-style=gnu --build-id --eh-frame-hdr -m elf_x86_64 -dynamic-linker /lib64/ld-linux-x86-64.so.2 -o bin/unittest-staged-fltree /usr/lib/x86_64-linux-gnu/crt1.o /usr/lib/x86_64-linux-gnu/crti.o /usr/bin/../lib/gcc/x86_64-linux-gnu/9/crtbegin.o -L/usr/bin/../lib/gcc/x86_64-linux-gnu/9 -L/lib/x86_64-linux-gnu -L/lib/../lib64 -L/usr/lib/x86_64-linux-gnu -L/usr/lib/llvm-13/bin/../lib -L/lib -L/usr/lib --export-dynamic src/test/crimson/seastore/onode_tree/CMakeFiles/unittest-staged-fltree.dir/test_staged_fltree.cc.o src/test/crimson/seastore/onode_tree/CMakeFiles/unittest-staged-fltree.dir/__/__/gtest_seastar.cc.o -rpath /opt/ceph/lib/x86_64-linux-gnu lib/libgmock_main.a lib/libgmock.a lib/libgtest.a -lpthread -ldl lib/libcrimson-seastore.a lib/libcrimson.a lib/libcrimson-common.a lib/libcrc32.a lib/libarch.a /opt/ceph/lib/x86_64-linux-gnu/libboost_iostreams.so.1.75.0 -lbz2 -llzma -lz -lzstd /opt/ceph/lib/x86_64-linux-gnu/libboost_random.so.1.75.0 lib/libjson_spirit.a lib/libcommon_utf8.a /usr/lib/x86_64-linux-gnu/libcrypto.so lib/libseastar.a -ldl /opt/ceph/lib/x86_64-linux-gnu/libboost_program_options.so.1.75.0 /opt/ceph/lib/x86_64-linux-gnu/libboost_thread.so.1.75.0 -lpthread /usr/lib/x86_64-linux-gnu/libcares.so /usr/lib/x86_64-linux-gnu/libcryptopp.so /usr/lib/x86_64-linux-gnu/libfmt.a /usr/lib/x86_64-linux-gnu/liblz4.so /usr/lib/x86_64-linux-gnu/libgnutls.so -latomic /usr/lib/x86_64-linux-gnu/libsctp.so -lrt /usr/lib/x86_64-linux-gnu/libyaml-cpp.so /usr/lib/x86_64-linux-gnu/libnuma.so -lstdc++ -lm -lgcc_s -lgcc -lc -lgcc_s -lgcc /usr/bin/../lib/gcc/x86_64-linux-gnu/9/crtend.o /usr/lib/x86_64-linux-gnu/crtn.o
```

**Comment by tchaikov** [comment]
jenkins test make check arm64

File: cmake/modules/BuildFIO.cmake [hunk 1]
```diff
@@ -13,10 +13,10 @@ function(build_fio)
   file(MAKE_DIRECTORY ${source_dir})
   ExternalProject_Add(fio_ext
     UPDATE_COMMAND "" # this disables rebuild on each run
-    GIT_REPOSITORY "https://github.com/axboe/fio.git"
+    GIT_REPOSITORY "https://github.com/ceph/fio.git"
```
Comment (rzarzynski) [inline cmake/modules/BuildFIO.cmake:16]: ACK.

File: cmake/modules/BuildDPDK.cmake [hunk 2]
```diff
@@ -82,11 +82,19 @@ function(do_build_dpdk dpdk_dir)
     set(dpdk_source_dir ${CMAKE_SOURCE_DIR}/src/spdk/dpdk)
   endif()
 
+  set(extra_cflags "-fPIC")
```
Comment (rzarzynski) [inline cmake/modules/BuildDPDK.cmake:85]: OK, `-fPIC` is nothing new.

**Review by rzarzynski — APPROVED** [review]

**Pull request merged**